app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -151,4 +167,88 @@ class CreditCardsSettingFragment : PreferenceFragmentCompat() {
 
         isCreditCardsListLoaded = true
     }
+
+    /**
+     * Show a warning to set up a pin/password when the device is not secured. This is only used
+     * when BiometricPrompt is unavailable on the device.
+     */
+    override fun showPinDialogWarning(context: Context) {
+        AlertDialog.Builder(context).apply {
+            setTitle(getString(R.string.credit_cards_warning_dialog_title))
+            setMessage(getString(R.string.credit_cards_warning_dialog_message))
+
+            setNegativeButton(getString(R.string.credit_cards_warning_dialog_later)) { _: DialogInterface, _ ->
+                navigateToCreditCardManagementFragment()
+            }
+
+            setPositiveButton(getString(R.string.credit_cards_warning_dialog_set_up_now)) { it: DialogInterface, _ ->
+                it.dismiss()
+                val intent = Intent(Settings.ACTION_SECURITY_SETTINGS)
+                startActivity(intent)
+            }
+
+            create()
+        }.show().secure(activity)
+    }
+
+    /**
+     * Create a prompt to confirm the device's pin/password and start activity based on the result.
+     * This is only used when BiometricPrompt is unavailable on the device.
+     *
+     * @param manager The device [KeyguardManager]
+     */
+    @Suppress("Deprecation")
+    override fun showPinVerification(manager: KeyguardManager) {
+        val intent = manager.createConfirmDeviceCredentialIntent(
+            getString(R.string.credit_cards_biometric_prompt_message_pin),
+            getString(R.string.credit_cards_biometric_prompt_message)
+        )
+        startActivityForResult(intent, BiometricPromptPreferenceFragment.PIN_REQUEST)
+    }
+
+    /**
+     * Navigates to the [CreditCardsManagementFragment] with a slight delay.
+     */
+    private fun navigateToSavedCreditCards() {
+        runIfFragmentIsAttached {
+            viewLifecycleOwner.lifecycleScope.launch(Dispatchers.Main) {
+                // Workaround for likely biometric library bug
+                // https://github.com/mozilla-mobile/fenix/issues/8438
+                delay(SHORT_DELAY_MS)
+                navigateToCreditCardManagementFragment()
+            }
+        }
+    }
+
+    /**
+     * Called when authentication succeeds.
+     */
```

Comment:
I would probably just remove this comment since the others don't have one either, and it's also not fully correct since it's also called on the negative button when we show the pin dialog warning.

Reply:
good catch. this was copy/paste from logins